### PR TITLE
[create_timepoint] Fix null project bug

### PR DESCRIPTION
**Pull request by ridz1208**
## Brief summary of changes
Fixed bug below and changed behaviour to only display projects user is affiliated with

https://github.com/aces/Loris/issues/5418

### CODE REVIEW

File: modules/create_timepoint/php/create_timepoint.class.inc [hunk 1]
```diff
@@ -266,11 +283,11 @@ class Create_Timepoint extends \NDB_Form
         }
 
         //Project
-        $projects = \Utility::getProjectList();
-        if (count($projects) == 0) {
-            $errors['project'] = "No projects are defined in the database, please ".
-            "add projects in order to continue";
-        } else if (count($projects) > 1 && is_null($val['project'])) {
+        $user_list_of_projects = $user->getData('ProjectIDs');
```
Comment (johnsaigle) [inline modules/create_timepoint/php/create_timepoint.class.inc]:
```suggestion
        $user_list_of_projects = $user->getProjectIDs());
```

File: modules/create_timepoint/php/create_timepoint.class.inc [hunk 2]
```diff
@@ -199,42 +201,57 @@ class Create_Timepoint extends \NDB_Form
...
+                        'psc',
+                        'Site selection is required',
+                        'required'
                     );
-                    if (!is_null($center)) {
-                        $psc_labelOptions[$siteID] = $center['Name'];
+                }
+
+                // List projects
+                $user_list_of_projects = $user->getData('ProjectIDs');
```
Comment (jesscall) [inline modules/create_timepoint/php/create_timepoint.class.inc]:
```suggestion
                $user_list_of_projects = $user->getProjectIDs();
```

File: modules/create_timepoint/php/create_timepoint.class.inc [hunk 3]
```diff
@@ -199,42 +201,57 @@ class Create_Timepoint extends \NDB_Form
...
-            if ($num_sites >1) {
-                $this->tpl_data['pscLabelAdded'] = true;
-                $psc_labelOptions = array(null => '');
-                foreach ($user_list_of_sites as $key => $siteID) {
-                    $center = $DB->pselectRow(
-                        "SELECT CenterID as ID, Name FROM psc 
+                $this->addRule('visitLabel', 'Visit label is required', 'required');
+
+                // List of sites for the user
+                $user_list_of_sites = $user->getData('CenterIDs');
```
Comment (jesscall) [inline modules/create_timepoint/php/create_timepoint.class.inc]:
```suggestion
                $user_list_of_sites = $user->getCenterIDs();
```

**Comment by driusan** [comment]
@ridz1208 travis

**Comment by ridz1208** [comment]
I used to have a way with Travis... not anymore

**Pull request merged**